[13.0][IMP] ddmrp improve po line on hand status

TDu opened the pull request:
Add the quantity ordered
Allow multiple edit
Allow editing scheduled date

TDu committed:
[13.0][IMP] ddmrp improve po line on hand status

Add the quantity ordered
Allow multiple edit
Allow editing scheduled date

jgrandguillaume reviewed (approved):
Functional test ok. Very useful change to allow people to move date of reception at line level. Works very nicely with https://github.com/OCA/purchase-workflow/tree/13.0/purchase_delivery_split_date

Business case: Vendore callss to move some position delivery and you need to update the reception date. Using the above module, it work so well and nicely.

LoisRForgeFlow committed:
[FIX] ddmrp: ddmrp_comment should be editable in PO line.

This has probably been missing in migration to v13.

LoisRForgeFlow commented:
This is indeed a good UX improvement!

While I was reviewing I realized the follow-up notes were not editable. I have opened a PR to include the fix here: https://github.com/camptocamp/ddmrp/pull/1.

jbaudoux commented on ddmrp/views/purchase_order_line_view.xml:
```diff
@@ -19,14 +20,15 @@
                     readonly="1"
                 />
-                <field name="date_planned" widget="date" readonly="True" />
+                <field name="date_planned" widget="date" />
```
Is it timezone safe having widget="date" when non readonly? On the PO editable tree view, there is no widget="date"

TDu replied:
@jbaudoux I don't know what you mean ? Could you explain ?

LoisRForgeFlow replied:
I think he means that since the field is a `Datetime`, the fact of using the widget `date` could lead to unexpected results. 

I have tested it and indeed weird thing can happen: if you select say "2021-05-25" you are writing "2021-05-25 00:00 UTC" in the database, and if you are in a "negative" timezone like the US, you will see that after saving the date, it is displayed as "2021-05-24" (conversion to your timezone of the previous UTC datetime).

See below a gif of my testing:
![Peek 2021-05-05 11-09](https://user-images.githubusercontent.com/23449160/117119635-a3cec680-ad92-11eb-8671-2f7c5a896649.gif)

Nice catch @jbaudoux !!

What should we do? Just removing the widget might be the easiest...

simahawk commented:
@TDu can you check travis? Seems unrelated to your changes tho.

TDu commented:
I rebased, Travis is :green_apple:

TDu committed:
[13.0][IMP] ddmrp improve po line on hand status

Remove the restricting domain showing only lines with a buffer. And
replace it with a default filter.

Finally remove the widget date which is not timezone friendly

LoisRForgeFlow commented on ddmrp/views/purchase_order_line_view.xml:
```diff
@@ -40,6 +40,11 @@
             <filter name="hide_cancelled" position="after">
+                <filter
+                    name="hide_no_buffer"
+                    string="Hide lines without buffer"
```
I think just "Buffered" will be a better name for the filter.

TDu replied:
Yes, changed it.

LoisRForgeFlow commented:
/ocabot merge minor
